RESTService/pyminsky.cc

```diff
@@ -30,28 +30,73 @@ namespace pyminsky
 
   minsky::Item& findObject(const std::string& typeName)
   {
-    using namespace minsky;
-    auto& canvas = minsky.canvas;
+      using namespace minsky;
+      auto& canvas = minsky.canvas;
 
-    canvas.item.reset();
+      canvas.item.reset(); // Reset the item to ensure no leftover references
 
-    if (typeName == "Group" && !canvas.model->groups.empty())
-      canvas.item = canvas.model->groups.front();
-    else
-      minsky.model->recursiveDo(&GroupItems::items, [&](const Items&, Items::const_iterator i)
+      bool found = false;
```

Comment:
This found flag is redundant. It just shadows the validity status of canvas.item

Reply:
Understood. I was getting a segfault in the renameVariables.sh test, so I used it for added safety.